elements/pfe-card/src/pfe-card.js

```diff
@@ -38,6 +39,67 @@ class PfeCard extends PFElement {
   constructor() {
     super(PfeCard);
   }
+
+  connectedCallback() {
+    super.connectedCallback();
+    let haxProps = {
```

Comment:
@btopro I'm hesitant to build HAX into the components directly because it seems like each CMS will have their own implementation requirements that might vary.  @mwcz was talking about adding a hook into the connectedCallback to allow things to be attached separately; that means, in theory, we could build out a HAX implementation for each CMS separately and hook them into the components in post.

Reply:
@starryeyez024 What do you think of this idea?  RH.com could have a HAX JS file that is a compilation of our CMS definitions for each component.  CP could have their own, etc.  So we'd include:
```
- pfelement.js
- pfe-card.js
- rdc-hax.js
```

Reply:
Another possible solution would be another custom element that you pile the definitions into and if THAT tag is appended then it fires the correct events to supply the CMS w/ those definitions that make the most sense for it. [Custom element registry](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/get) could be used to test that a tag's been defined and then broadcast as another way of keeping things tidy / abstracted per implementation

This PR was mostly so our talk has some fodder of "look what happens when we all adopt the platform" to ensure this isn't accepted accidentally :)

Reply:
@starryeyez024 thank you for the context. Speaking to the options B vs C which could provide a nice happy medium in flexibility vs constraints...

A paradigm that @nikkimk has illustrated w/ [a11y-media-player](https://www.webcomponents.org/element/@lrnwebcomponents/a11y-media-player) is that she's made a pretty ridiculously flexible / super-power laden video player. It's WAAAYYYY too much horsepower for a typical user that would be using a UI to implement. So, we use that element in our templates that are delivered by our media system (Drupal based). This allows us to wire up highly complex features of the player from our content type / fields, leveraging slots etc.

For end users / HAX / content blob areas they get the [video-player tag](https://github.com/elmsln/lrnwebcomponents/blob/master/elements/video-player/video-player.js#L63-L93). The `video-player` tag then leverages a11y-media-player so we get design consistency but we allow HAX and the wrapper tag to provide the constrains needed to make it manageable for end-users.

A happy medium toward the B/C option could also be making semantically named tags (splash-card instead of pfe-card sorta deal) that have the HAX wiring in them while the raw tags currently in patternfly are the most flexible for other devs to implement consistency in design / theme layer.

Reply:
@castastrophe @mwcz could we include the haxProps in the connected callback of each of our content & container components? The combos might be the only ones that need site-specific definitions for the fields.

Reply:
as an FYI bc of this thread I've recently enhanced the way we can detect if something is HAX capable -- https://github.com/elmsln/lrnwebcomponents/blob/master/elements/hax-body/lib/hax-store.js#L479-L492

english translation - If one of the 3 common methods of implementing HAX into an element is detected, load the definition without needing to fire the connected callback. Fallback on the connected callback for older elements.

This was in part for performance as without having to connect it to the DOM we can get the definitions faster, especially as we move closer to dynamic imports for all with no polyfill or compiling, it'll be lightning compared to having to connect and disconnect every asset at run time.

This code is running on haxtheweb.org and has reduced the time to boot up HAX as a result

Reply:
I'm for adding HAX-compatibility as a "plugin" to each PF Element, but I don't want to include HAX properties in the original source of the elements.  I implemented #270 so that you can add onConnectedCallback-type stuff to an element without modifying its original source.  Since connecting to the DOM isn't required anymore, we can do something like these steps:

 1. include `pfe-card.js`
 2. include `pfe-card.hax.js` which adds haxProperties to the PfeCard class
 3. start creating `<pfe-card>` elements